vcsclient/azurerepos.go

```diff
@@ -568,20 +568,44 @@ func parsePullRequestDetails(pullRequest git.GitPullRequest, repository string)
 		prBody = *bodyPtr
 	}
 
+	// When a pull request is from a forked repository,extract the owner.
+	sourceRepoOwner := owner
+	if pullRequest.ForkSource != nil {
+		if sourceRepoOwner = extractOwnerFromForkedRepoUrl(pullRequest.ForkSource); sourceRepoOwner == "" {
+			client.logger.Warn(failedForkedRepositoryExtraction)
+		}
+	}
+
 	return PullRequestInfo{
 		ID:   int64(*pullRequest.PullRequestId),
 		Body: prBody,
 		Source: BranchInfo{
 			Name:       shortSourceName,
 			Repository: repository,
+			Owner:      sourceRepoOwner,
 		},
 		Target: BranchInfo{
 			Name:       shortTargetName,
 			Repository: repository,
+			Owner:      owner,
 		},
 	}
 }
 
+// Extract the repository owner of a forked source
+func extractOwnerFromForkedRepoUrl(forkedGit *git.GitForkRef) string {
+	if forkedGit == nil || forkedGit.Repository == nil || forkedGit.Repository.Url == nil {
+		return ""
+	}
+	url := *forkedGit.Repository.Url
+	owner := strings.Split(strings.TrimPrefix(url, "https://dev.azure.com/"), "/")[0]
+	// Failed to properly extract owner name
+	if strings.Contains(owner, "http") {
+		return ""
+	}
```

Comment:
If we fail why the owner string will contain the "http"? Can you elaborate on this scenario?

Reply:
You're right, it wasn't clear. i've changed it 👍